### ISSUE TITLE
.NET Standard 2.0 support (FileNotFound exception)

### ISSUE THREAD

**Issue by tomerdomshlak** [mono/Embeddinator-4000#518]
Hi,

I've tried to compile the latest version on Windows with those steps:

1. clone the repository.
2. run "git submodule update --init --recursive".
3. open the solution.
4. restore nuget packages.
5. build.

I'm having the following compilation errors:

![image](https://user-images.githubusercontent.com/20470843/31129649-e388ed3e-a85d-11e7-893d-e9073a9d9f79.png)

Any clue what I'm doing wrong?

Thanks!

**Comment by tritao** [comment]
Can you try calling building with our `./build.ps1` script?

**Comment by tomerdomshlak** [comment]
Hi,

Thanks.

I've run ./build.ps1 and it completed successfully.

Then, I've entered the solution and tried to build it.

All projects compiled successfully besides "managed-macos-full", "managed-macos-system", "objcgentest" and "objcgen".

I've unloaded those projects (as I'm trying to compile an Android AAR), and the solution compiled successfully.

Then, I've navigated to "Embeddinator-4000\build\lib\Debug", and run the following command:

./Embeddinator-4000.exe --gen=Java --platform=Android -c -v "**folderPath**\MonoTestLibrary.dll"

It failed with the following output:

`PS C:\Workspace\Embeddinator-4000\build\lib\Debug> ./Embeddinator-4000.exe --gen=Java --platform=Android -c -v "C:\Users
\tomer\Source\Repos\MonoTestLibrary\MonoTestLibrary\bin\Debug\netstandard2.0\MonoTestLibrary.dll"
Parsing assemblies...
      Key HKCU\SOFTWARE\Novell\Mono for Android\AndroidSdkDirectory not found.
      Key HKCU\SOFTWARE\Novell\Mono for Android\AndroidNdkDirectory not found.
      Key HKCU\SOFTWARE\Novell\Mono for Android\JavaSdkDirectory not found.
    Looking for Android SDK..
      Key HKCU\SOFTWARE\Novell\Mono for Android\AndroidSdkDirectory not found.
      Key HKLM\SOFTWARE\Novell\Mono for Android\AndroidSdkDirectory not found.
      Key HKCU\SOFTWARE\Xamarin\MonoAndroid\PrivateAndroidSdkPath not found.
      Key HKCU\SOFTWARE\Android SDK Tools\Path not found.
      Key HKLM\SOFTWARE\Android SDK Tools\Path found:
    Path contains adb in \platform-tools (C:\Program Files (x86)\Android\android-sdk).
    Looking for Android NDK..
      Key HKCU\SOFTWARE\Novell\Mono for Android\AndroidNdkDirectory not found.
      Key HKLM\SOFTWARE\Novell\Mono for Android\AndroidNdkDirectory not found.
      Key HKCU\SOFTWARE\Novell\Mono for Android\JavaSdkDirectory not found.
      Key HKLM\SOFTWARE\Novell\Mono for Android\JavaSdkDirectory not found.
    Looking for Java 6 SDK..
      Key HKLM\SOFTWARE\JavaSoft\Java Development Kit\CurrentVersion found.
      Key HKLM\SOFTWARE\JavaSoft\Java Development Kit\1.6\JavaHome not found.
      Key HKLM\SOFTWARE\JavaSoft\Java Development Kit\1.7\JavaHome not found.
      Key HKLM\SOFTWARE\JavaSoft\Java Development Kit\1.8\JavaHome found:
    Path contains jarsigner.exe in \bin (C:\Program Files (x86)\Java\jdk1.8.0_112).
    Found Android SDK. API levels: 19, 20, 21, 22, 23, 24
    Parsed 'MonoTestLibrary.dll'
Processing assemblies...

Unhandled Exception: System.IO.FileNotFoundException: netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b
13ffcd2ddd51
   at IKVM.Reflection.Universe.Load(String refname, Module requestingModule, Boolean throwOnError) in C:\Workspace\Embed
dinator-4000\external\ikvm\reflect\Universe.cs:line 756
   at IKVM.Reflection.Reader.ModuleReader.ResolveAssemblyRefImpl(Record& rec) in C:\Workspace\Embeddinator-4000\external
\ikvm\reflect\Reader\ModuleReader.cs:line 594
   at IKVM.Reflection.Reader.ModuleReader.ResolveAssemblyRef(Int32 index) in C:\Workspace\Embeddinator-4000\external\ikv
m\reflect\Reader\ModuleReader.cs:line 577
   at IKVM.Reflection.Reader.ModuleReader.ResolveType(Int32 metadataToken, IGenericContext context) in C:\Workspace\Embe
ddinator-4000\external\ikvm\reflect\Reader\ModuleReader.cs:line 460
   at IKVM.Reflection.Reader.TypeDefImpl.get_BaseType() in C:\Workspace\Embeddinator-4000\external\ikvm\reflect\Reader\T
ypeDefImpl.cs:line 59
   at IKVM.Reflection.Type.get_IsEnum() in C:\Workspace\Embeddinator-4000\external\ikvm\reflect\Type.cs:line 1335
   at Embeddinator.Generators.ASTGenerator.Visit(TypeInfo typeInfo) in C:\Workspace\Embeddinator-4000\binder\Generators\
AstGenerator.cs:line 105
   at Embeddinator.Generators.ASTGenerator.Visit(Assembly assembly) in C:\Workspace\Embeddinator-4000\binder\Generators\
AstGenerator.cs:line 68
   at Embeddinator.Driver.Process() in C:\Workspace\Embeddinator-4000\binder\Driver.cs:line 80
   at Embeddinator.Driver.Run() in C:\Workspace\Embeddinator-4000\binder\Driver.cs:line 236
   at Embeddinator.CLI.Main(String[] args) in C:\Workspace\Embeddinator-4000\binder\CLI.cs:line 215`

Should I install the Android NDK? Any idea?

Thanks.

**Comment by tritao** [comment]
Looks to be some issue related to loading this `netstandard.dll` file... I assume it's related to .NET Standard 2.0 assemblies. I will need to check this.

**Comment by tomerdomshlak** [comment]
Awesome.

BTW,

![image](https://user-images.githubusercontent.com/20470843/31137740-c6a8ad96-a874-11e7-81d2-26d4651f9dc2.png)

This is on the same DLL only .NET Standard 1.6.

**Comment by jahmai** [comment]
This project is really exciting but this issue is a blocker for it being useful for us. +1!

**Comment by tritao** [comment]
I'm trying to reproduce this, but so far cannot do it on macOS with Mono 5.4, either for .NET Standard 1.6 or 2.0. It might be a Windows-only issue.

**Comment by jonathanpeppers** [comment]
Hi @tomerdomshlak this issue hasn't had any activity in a little while, but I was able to run Embeddinator on Windows against a netstandard 1.6 library and output an Android AAR. It was [this](https://github.com/mono/Embeddinator-4000/blob/master/tests/managed/netstandard/managed-netstandard.csproj) project.

I have Visual Studio 2017 15.6.3, and I was using Embeddinator-4000 built from source. Could your issue have been resolved on master now?

Closing for now, feel free reopen if you have more details, etc.
